testing/regress/hpcc/regression/cleanup.py

```diff
@@ -0,0 +1,102 @@
+    for logFilePath in regressLogFiles:
+        logFileNamePattern = r'^(thor|roxie|hthor)\.\d{2}-\d{2}-\d{2}-\d{2}-\d{2}-\d{2}\.log(?!.*exclusion)'
+        match = re.search(logFileNamePattern, os.path.basename(logFilePath))
+        if match:
+            if mode == "all" or mode == "workunits":
+                getWorkunitDetails(logFilePath)
+            elif mode == "passed":
+                getPassedWorkunitDetails(logFilePath)
+
+# Extracts workunit details from log file
+def getWorkunitDetails(logFilePath):
+    with open(logFilePath, 'r') as file:
+        lineList = file.readlines()
+        for line in lineList:
+            index = line.find("URL")
+            if index != -1:
+                url = line[index + 3:-1]
+                startIndex = url.find('Wuid=')
+                if startIndex != -1:
+                    wuid = url[startIndex:]
+                    deleteWorkunit(url, wuid)
+
+# Extracts workunit details from log file for passed test cases
+def getPassedWorkunitDetails(logFilePath):
+    with open(logFilePath, 'r') as file:
+        lineList = file.readlines()
+        for i, line in enumerate(lineList):
+            if re.search(r'\bpass\b', line, re.IGNORECASE):
+                nextLine = lineList[i + 1]
+                index = nextLine.find("URL")
+                if index != -1:
+                    url = nextLine[index + 3:-1]
+                    startIndex = url.find('Wuid=')
+                    if startIndex != -1:
+                        wuid = url[startIndex:]
+                        deleteWorkunit(url, wuid)
+
+# Deletes workunits
+def deleteWorkunit(url, wuid):
+    deletionUrl = url.replace("?Widget=WUDetailsWidget&Wuid", "WsWorkunits/WUDelete.json?Wuids")
+    try:
+        response = requests.post(deletionUrl)
+        jsonResponse = response.json()
+        if response.status_code == 200:
+            if jsonResponse.get("WUDeleteResponse") == {}:
+                logger.info("Workunit %s deleted successfully.", wuid)
+            elif "ActionResults" in jsonResponse["WUDeleteResponse"]:
+                errorMessage = jsonResponse["WUDeleteResponse"]["ActionResults"]["WUActionResult"][0]["Result"]
+                logger.error("Failed to delete workunit %s.\n   URL:%s\n   %s Response status code: %d", wuid, url, errorMessage, response.status_code)
+        else:
+            logger.error("Failed to delete workunit %s.\n   URL:%s\n   Response status code: %d", wuid, url, response.status_code)
+    except requests.exceptions.RequestException as e:
+        logger.error("Error occurred while deleting workunit %s: %s.\n   URL: %s", wuid, str(e), url)
+
+# Main function to instantiate custom logger and initiate workunit detail extraction
+def checkCleanupMode(cleanupMode):
+    configData = loadConfig('ecl-test.json')
+    logDirPath = os.path.expanduser(configData['Regress']['logDir'])
+    buildCleanupLogger(logDirPath, logger)
+
+    if cleanupMode in ['all', 'passed', 'workunits']:
```

Comment:
This condition is already checked in the ecl-test, why it is need here?

Reply:
Sorry for the oversight, we realised that the if condition is not required here.